cmd/secret-collection-manager/main.go

```diff
@@ -0,0 +1,415 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/julienschmidt/httprouter"
+	uuid "github.com/satori/go.uuid"
+	"github.com/sirupsen/logrus"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/test-infra/prow/logrusutil"
+
+	"github.com/openshift/ci-tools/pkg/vaultclient"
+)
+
+const objectPrefix = "secret-collection-manager-managed"
+
+type option struct {
+	// Folder under which to create policies
+	kvStorePrefix string
+	listenAddr    string
+	vaultAddr     string
+	vaultToken    string
+}
+
+func parseOptions() *option {
+	o := &option{}
+	flag.StringVar(&o.kvStorePrefix, "kv-store-prefix", "secret/self-managed", "Vault KV folder under which all policies will get created")
+	flag.StringVar(&o.listenAddr, "listen-addr", "127.0.0.1:8080", "The address to listen on")
+	flag.StringVar(&o.vaultAddr, "vault-addr", "http://127.0.0.1:8300", "The address under which vault should be reached")
+	flag.StringVar(&o.vaultToken, "vault-token", "jpuxZFWWFW7vM882GGX2aWOE", "The privileged token to use when communicating with vault, must be able to CRUD policies")
+	flag.Parse()
+	return o
+}
+
+func main() {
+	logrusutil.ComponentInit()
+	o := parseOptions()
+
+	privilegedVaultClient, err := vaultclient.New(o.vaultAddr, o.vaultToken)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to construct vault client")
+	}
+
+	if err := server(privilegedVaultClient, o.kvStorePrefix, o.listenAddr).ListenAndServe(); err != nil && err != http.ErrServerClosed {
+		logrus.WithError(err).Fatal("failed to listen and serve")
+	}
+
+}
+
+func server(privilegedVaultClient *vaultclient.VaultClient, kvStorePrefix, listenAddr string) *http.Server {
```

Comment:
What's the use of breaking this func out?

Reply:
That func is what the tests use